cloudinit/net/network_manager.py

```diff
@@ -105,6 +107,15 @@ def _set_ip_method(self, family, subnet_type):
         if self.config[family]["method"] == "auto" and method == "manual":
             return
 
+        flavor = self.cloud_config.get("flavor", "rhel")
```

Comment:
I'm not sure that checking for flavor here really makes any sense given that [RHEL doesn't provide any network-manager specific renderer configuration](https://github.com/canonical/cloud-init/blob/main/cloudinit/distros/rhel.py#L33) (nor does any other distro ATM). It seems to give the illusion that this would be a RHEL-only change, rather than a change affecting all distros using network-manager. If you want it to be RHEL-only, then you need `network-manager` config in the renderer configuration. However, given the nature of this change, I think it makes sense to apply it universally.

Reply:
Oh I see. I was looking at [this](https://github.com/canonical/cloud-init/blob/main/cloudinit/net/sysconfig.py#L440) and I thought I add that check because it would simply make it symmetric with sysconfig.

Reply:
If you think applying it universally would be safe and we would not break anyone else, I am fine with removing the check.

Reply:
Yes, in this case I think it should be fine.

Reply:
Addressed in latest patch.